File: runner/examples/list-executors.ts
// Run with 'npx ts-node src/test-client.ts'

import runnerClient from '../src/server/services/runner/runner-client';

void (async function main () {
  runnerClient.ListExecutors({}, (err, response) => {
    if (err) {
      console.error('List request error: ', err);
    } else {
      console.log('list response: ', JSON.stringify({ response }, null, 2));
    }
  });
})();

File: runner-client/examples/list_executors.rs
use tonic::Request;

use runner::runner_client::RunnerClient;
use runner::ListExecutorsRequest;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut client = RunnerClient::connect("http://localhost:7001").await?;

    let response = client
        .list_executors(Request::new(ListExecutorsRequest {}))
        .await?;

    println!("{:#?}", response.into_inner());

    Ok(())
}
